Tests/KituraWebSocketTests/WebSocketClient.swift

```diff
@@ -156,3 +157,33 @@ class HTTPClientHandler: ChannelInboundHandler, RemovableChannelHandler {
 extension HTTPVersion {
     static let http11 = HTTPVersion(major: 1, minor: 1)
 }
+
+protocol WebSocketDelegate {
```

Comment:
I suggest we call this `WebSocketClientDelegate`. Also, please document what each of the member signifies and when it will be invoked.